dir.props

```diff
@@ -241,6 +241,9 @@
   <PropertyGroup Condition="'$(Platform)' == 'arm64'">
     <DefineConstants>ARM64;BIT64;$(DefineConstants)</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Platform)' == 'wasm'">
+    <DefineConstants>WASM;BIT32;PLATFORM_UNIX;$(DefineConstants)</DefineConstants>
```

Comment:
Treating WASM as Unix flavor makes sense. We should do it consistently, e.g. set `TargetsUnix` for WASM as well.

Reply:
That seems to break nuget restore (at least while building on Windows). I agree the extra msbuild definitions aren't ideal, but is it ok to keep them until we have wasm nuget packages?

Reply:
If it breaks nuget restore, make a specific workaround for WASM nuget restore. I think it should be in src\Framework\Framework-native.depproj - do not pull in `Microsoft.NETCore.Native` for WASM.

Reply:
Done. What are we missing?

Reply:
> What are we missing by not pulling that in?

Nothing really, because there's no package for WASM anyway. It's all of this: https://github.com/dotnet/corefx/tree/c280881a048ee9d9fbfc629ca55d755d3e2b045d/src/Native/Unix/System.Native

The native library supports CoreFX managed assemblies that need to interact with the OS.